test/unit_tests/utils/test_search_query.py

```diff
@@ -102,3 +104,28 @@ def test_query_kv_camelcasing():
     q = "productTag:myTag"
     q2 = "product_tag:myTag"
     assert _parse_tree_and_tsquery(q)[1] == _parse_tree_and_tsquery(q2)[1]
+
+
+@pytest.mark.parametrize(
+    "search_query,substrings",
+    [
+        ("name:Bob", ["WHERE my_table.name LIKE 'bob'"]),
+        ("empty:column", ["WHERE false"]),
+        ("bob| joe", ["WHERE my_table.name LIKE 'bob'", "UNION SELECT", "WHERE my_table.name LIKE 'joe'"]),
+        ("(bob) | -(joe)", ["WHERE my_table.name LIKE 'bob'", "UNION SELECT", "WHERE my_table.name LIKE 'joe'"]),
+        ("-(joe)", ["WITH", "LEFT OUTER JOIN", ".id IS NULL"]),
+    ],
+)
+def test_sqlalchemy_visitor(search_query, substrings):
+    class MyTable(BaseModel):
+        __tablename__ = "my_table"
+        id = Column(Integer, primary_key=True)
+        name = Column(String)
+
+    mappings = {"name": lambda node: MyTable.name.like(node[1].lower())}
+    stmt = select(MyTable)
+    stmt = create_sqlalchemy_select(stmt, search_query, mappings, MyTable, MyTable.id)
```

Comment:
Every time you reuse a variable somewhere on this world a baby seal dies ;)

Reply:
Generally I agree but for query building I don't see much benefit in creating new variables that you'll never need

i.e.

```
stmt_start = select(MyTable)
stmt_with_select = create_sqlalchemy_select(stmt_start)
stmt_with_stuff = add_more_stuff(stmt_with_select)
stmt_with_more = even_more(stmt_with_stuff) if something else stmt_with_stuff
...
```

vs

```
stmt = select(MyTable)
stmt = create_sqlalchemy_select(stmt)
stmt = add_more_stuff(stmt)
if something:
  stmt = even_more(stmt)
```

Might just be an inherent problem with using an ORM to create dynamic queries